pkg/archive/install.go

```diff
@@ -31,7 +31,7 @@ func DownloadToGopathBin(opts DownloadArchiveOptions) error {
 	// determine the appropriate file extension based on the OS, e.g. windows gets .zip, otherwise .tgz
 	opts.Ext = opts.ArchiveExtensions[runtime.GOOS]
 	if opts.Ext == "" {
-		return errors.Errorf("no archive file extension was specified for the current GOOS (%s)", runtime.GOOS)
+		return fmt.Errorf("no archive file extension was specified for the current GOOS (%s)", runtime.GOOS)
```

Comment:
Lines like this alter the original logic of errors.Errorf, where if the wrapped error is nil, then Errorf returns nil as well. 

When switching to fmt.Errorf, you need to check if the error is not nil and then call fmt.Errorf, otherwise returning nil.

If you look for return errors.Errorf, all of those statements need a conditional in your PR.

Reply:
Makes sense, I'll change it as suggested :+1: